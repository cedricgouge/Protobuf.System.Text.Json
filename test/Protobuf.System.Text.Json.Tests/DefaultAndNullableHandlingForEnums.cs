using System.Text.Json;
using System.Text.Json.Protobuf.Tests;
using System.Text.Json.Serialization;
using Protobuf.System.Text.Json.Tests.Utils;
using SmartAnalyzers.ApprovalTestsExtensions;
using Xunit;

namespace Protobuf.System.Text.Json.Tests
{
    public class DefaultAndNullableHandlingForEnums
    {
        [Fact]
        public void Should_Remove_DefaultEnumValues()
        {
            var msg = new MessageWithEnum2
            {
                EnumField = TestEnum2.Unknown,
                BoolField = true,
                DoubleField = 1.0
            };

            var jsonSerializerOptions = TestHelper.CreateJsonSerializerOptions(conf => conf.UseStringProtoEnumValueNames = true);
            
            jsonSerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingDefault;

            // Act
            var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);

            // Assert
            var approver = new ExplicitApprover();
            approver.VerifyJson(serialized);
        }
    }
}
