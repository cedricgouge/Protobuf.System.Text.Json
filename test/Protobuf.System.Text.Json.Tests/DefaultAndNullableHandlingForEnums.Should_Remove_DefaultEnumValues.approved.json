{
  "boolField": true,
  "doubleField": 1
}